### DISCUSSION TITLE
importing 3rd party libs

### DISCUSSION THREAD

**Discussion by mblarsen** [google/zx#350] [Help needed]
I'm using `zx` mainly like this:

```js
#!/usr/bin/env zx

const config = YAML.parse((await $`cat config.yaml`).toString())
...
```

With `zx` installed globally. I `chmod +x` and run the script `./myscript` without ext.

It is not clear to me how I'm supposed to use 3rd party libs. E.g. want to make use of `keyblade`.

```js
#!/usr/bin/env zx

import {keyblade} from 'keyblade'

const config = keyblad(YAML.parse((await $`cat config.yaml`).toString()))
...
```

However, this causes an import error

```
Error [ERR_MODULE_NOT_FOUND]: Cannot find package 'keyblade' imported from [CUT] myscript.mjs
```

`keyblade` is installed globally as well.

Is what I'm trying to do not possible? Do I have to use the [_importing from other scripts_](https://github.com/google/zx#importing-from-other-scripts) approach to achieve this?

**Comment by antonmedv** [comment]
To import globally installed packages check your NODE_PARH env var.

**Comment by craftogrammer** [comment]
I have the same use case, I don't know how to do it so I forked the project, and added my dependencies on my fork, and it's working fine, for time being I found it easier.

After some time, I went ahead, and imported the `$` function from `zx`, and used pkg to convert the script into binary so on other user's PC,  they don't even have to install node or npm packages.

I find the `zx` implementation of running shell commands easier than the other node packages because I can write the whole shell command as it is, and I don't have to escape anything. It feels more natural.

Just sharing my experience here.

**Comment by antongolub** [comment]
A lot has changed, here's the actual status:
* CLI `--install` flag fetches deps before the script running.
https://google.github.io/zx/cli#install
https://github.com/google/zx/blob/e1b66cba78282bb6511fd768c5838d8eb9413bb7/test/deps.test.js#L118
* CLI `--prefer-local` accepts `node_modules` directory to load the deps from.
https://google.github.io/zx/cli#prefer-local-l
